CruCentralCoast/Community/CommunityGroups/CommunityGroupsVC.swift

```diff
@@ -48,29 +72,163 @@ class CommunityGroupsVC: UITableViewController {
             return 1
         }
     }
-
+    
+    override func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+        if isFiltering() {
+            print("filtered days: ", self.filteredDays)
+            return self.filteredDays[section].rawValue.capitalized
+        }
+        
+        return self.days[section].rawValue.capitalized
+    }
+    
+    override func numberOfSections(in tableView: UITableView) -> Int {
+        if isFiltering() {
+            return self.filteredDays.count
+        }
+        
+        return self.days.count
+    }
+    
     override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return self.dataArray.count
+        if isFiltering() {
+            let day = self.filteredDays[section]
+            if let filteredGroups = self.filteredDictionary[day] {
+                return filteredGroups.count
+            }
+        }
+        else {
+            let day = days[section]
+            if let communityGroups = self.dataDictionary[day] {
+                return communityGroups.count
+            }
+        }
+        
+        return 0
+
     }
     
     override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
         let cell = tableView.dequeueCell(CommunityGroupCell.self, indexPath: indexPath)
-        
-        cell.configure(with: self.dataArray[indexPath.row])
-
+        if isFiltering() {
+            // check if there's item(s) for that day
+            if let filteredDay = self.filteredDictionary[self.filteredDays[indexPath.section]] {
+                cell.configure(with: filteredDay[indexPath.row])
+            }
+        }
+        else {
+            // check if there's item(s) for that day
+            if let day = self.dataDictionary[self.days[indexPath.section]] {
+                cell.configure(with: day[indexPath.row])
+            }
+        }
+         
         return cell
     }
     
     override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
         let vc = UIStoryboard(name: "CommunityGroups", bundle: nil).instantiateViewController(CommunityGroupDetailsVC.self)
-        vc.configure(with: self.dataArray[indexPath.row])
+        if isFiltering() {
+            if let filteredDay = self.filteredDictionary[self.filteredDays[indexPath.section]] {
+                vc.configure(with: filteredDay[indexPath.row])
+            }
+        }
+        else {
+            if let day = self.dataDictionary[self.days[indexPath.section]] {
+                vc.configure(with: day[indexPath.row])
+            }
+        }
+        
         self.navigationController?.pushViewController(vc, animated: true)
     }
+    
+    
+    
+    // MARK: - Private instance methods
```

Comment:
If these are private methods, then they ought to have the keyword "private" in front of each method

Reply:
ya whoops thanks